packages/webcomponents/src/components/business-forms/payment-provisioning/document-uploads/business-document-upload-form-step.tsx

```diff
@@ -0,0 +1,236 @@
+import { Component, Host, h, Prop, State, Method, Event, EventEmitter } from '@stencil/core';
+import { FormController } from '../../../form/form';
+import { BusinessFormSubmitEvent, DocumentFormServerErrorEvent, DocumentFormServerErrors } from '../../utils/business-form-types';
+import { IBusiness } from '../../../../api/Business';
+import Api, { IApiResponse } from '../../../../api/Api';
+import { config } from '../../../../../config';
+import { businessDocumentSchema } from '../../schemas/business-document-upload-schema';
+import { FileSelectEvent } from '../../../../components';
+import { EntityDocumentType, EntityDocument, EntityDocumentStorage } from '../../../../api/Document';
+
+@Component({
+  tag: 'justifi-business-document-upload-form-step',
+  styleUrl: 'business-document-upload-form-step.scss',
+})
+export class BusinessDocumentFormStep {
+  @Prop() authToken: string;
+  @Prop() businessId: string;
+  @Prop() allowOptionalFields?: boolean;
+  @State() formController: FormController;
+  @State() errors: any = {};
+  @State() documents: any = [];
+  @State() documentData: EntityDocumentStorage = new EntityDocumentStorage();
+  @Event({ bubbles: true }) submitted: EventEmitter<BusinessFormSubmitEvent>;
+  @Event() formLoading: EventEmitter<boolean>;
```

Comment:
Beside keeping event names consistent, another good reason for using kebab-case on our event names is that I think it will be easier to track events automatically on Analytics in the future. 

```suggestion
  @Event({ eventName: 'form-loading' }) formLoading: EventEmitter<boolean>;
```